components/fonts/font_context.rs

```diff
@@ -653,40 +781,44 @@ impl RemoteWebFontDownloader {
                 match downloader.handle_web_font_fetch_message(response_message) {
                     DownloaderResponseResult::InProcess => {},
                     DownloaderResponseResult::Finished => {
-                        if !downloader.process_downloaded_font_and_signal_completion(&state) {
-                            downloader
-                                .font_context
-                                .process_next_web_font_source(state.clone())
+                        if !downloader.process_downloaded_font_and_signal_completion() {
+                            let state = downloader.take_state();
+                            font_context.process_next_web_font_source(state)
                         }
                     },
-                    DownloaderResponseResult::Failure => downloader
-                        .font_context
-                        .process_next_web_font_source(state.clone()),
+                    DownloaderResponseResult::Failure => {
+                        let state = downloader.take_state();
+                        font_context.process_next_web_font_source(state)
+                    },
                 }
             }),
         )
     }
 
+    fn take_state(&mut self) -> WebFontDownloadState {
+        self.state
+            .take()
+            .expect("must be non-None until download either succeeds or fails")
+    }
+
     /// After a download finishes, try to process the downloaded data, returning true if
     /// the font is added successfully to the [`FontContext`] or false if it isn't.
-    fn process_downloaded_font_and_signal_completion(&self, state: &WebFontDownloadState) -> bool {
-        if self
-            .font_context
-            .web_fonts
-            .read()
-            .font_load_cancelled_for_stylesheet(&state.stylesheet)
-        {
-            self.font_context
-                .handle_web_font_load_finished(&state.finished_callback, false);
+    fn process_downloaded_font_and_signal_completion(&mut self) -> bool {
+        let state = self
+            .state
+            .as_ref()
+            .expect("must be non-None until processing is completed");
+        if state.font_load_cancelled() {
+            self.take_state().handle_web_font_load_failure();
             // Returning true here prevents trying to load the next font on the source list.
             return true;
         }
 
-        let font_data = std::mem::take(&mut *self.response_data.lock());
-        trace!(
-            "@font-face {} data={:?}",
+        let font_data = std::mem::take(&mut self.response_data);
+        debug!(
+            "downloaded @font-face {} data={:?}",
             self.web_font_family_name,
-            font_data
+            font_data.len()
```

Comment:
Maybe this change is accidental?

Reply:
I noticed that this was actually dumping the data as a vector of bytes to the console, making the log unnecessarily huge and unreadable. Is it ever useful to log the raw bytes of the font?